torch/_dynamo/variables/torch.py

```diff
@@ -888,14 +896,45 @@ def speculate_subgraph(f, sub_args, graph_checkpoint, checkpoint):
         if self.value.__name__ == "cond":
             # TODO(voz): Support fake tensor dispatch for recursive
             # ops - see torch/dispatch/_dispatcher.py
-            assert len(args) == 4
-            assert type(args[0]) in (
-                TensorVariable,
-                SymNodeVariable,
-                ConstantVariable,
-            ), str(
-                type(args[0])
-            )  # predicate
+            if len(args) != 4:
+                raise UserError(
+                    UserErrorType.DYNAMIC_CONTROL_FLOW,
+                    f"Expected 4 arguments but got {len(args)}.\n"
+                    f"Usage: cond(pred, true_fn, false_fn, operands)",
+                )
+            # predicate
+            if type(args[0]) not in (TensorVariable, SymNodeVariable, ConstantVariable):
+                raise UserError(
+                    UserErrorType.DYNAMIC_CONTROL_FLOW,
+                    f"Expected pred to be traced as TensorVariable, "
+                    f"SymNodeVariable or ConstantVariable but got {str(type(args[0]))} "
```

Comment:
TensorVariable, SymNodeVariable and ConstantVariable are meaningless to end users. They aren't going to know anything about how Dynamo internally structures its VariableTracker hierarchy.

Reply:
Yeah, rephrase the error message in a more user understandable way